include/roboptim/core/differentiable-function.hh

```diff
@@ -252,6 +252,16 @@ namespace roboptim
 				const_argument_ref argument,
 				size_type functionId = 0)
       const = 0;
+
+  public:
+    /// \brief Get the type-checking flag
+    virtual unsigned long flag()
+    {
+      return flag_;
+    }
+
+    /// \brief Flag representing the Roboptim Function type
+    static const unsigned long flag_;
```

Comment:
No need for a trailing underscore for static parameters, since you call it in a public way, e.g. `GenericDifferentiableFunction<T>::flag_`.